torch/nn/parallel/distributed.py

```diff
@@ -309,10 +309,8 @@ class DistributedDataParallel(Module):
                    only be provided when the input module resides on a single
                    CUDA device. For single-device modules, the i'th
                    :attr:`module` replica is placed on ``device_ids[i]``. For
-                   multi-device modules and CPU modules, ``device_ids`` must be
-                   ``None`` or an empty list, and input data for the forward
-                   pass must be placed on the correct device. (default: all
-                   visible devices for single-device modules)
+                   CPU modules, ``device_ids`` must be ``None`` or an empty list.
+                   Multi-device module is no longer supported.
```

Comment:
I don't think this is correct, we still do want to support multi-device modules right? If a module spans across multiple GPUs (ex: pipeline across 4 GPUs) we still want DDP to support that. I believe what we want to deprecate is the mode where a single DDP process is driving replicas across multiple GPUs.

Reply:
Thanks for the explanation! Reverted the changes here.